project1/server/server.cpp

```diff
@@ -14,16 +14,16 @@ namespace server {
         std::shared_ptr<server_tasks::CommandIDs> active_ids;
 
         // pass active command list to nPortTask and tPortTask
-        auto nPortTask = std::make_shared<server_tasks::NPortTask>(active_ids,nPort,
-                                                                   read_manager_,write_manager_);
-        auto tPortTask = std::make_shared<server_tasks::TPortTask>(active_ids,tPort,
-                                                                   read_manager_,write_manager_);
+        auto nPortTask = std::make_shared<server_tasks::NPortTask>(std::move(active_ids),nPort,
+                                                                   std::move(read_manager_),std::move(write_manager_));
+        auto tPortTask = std::make_shared<server_tasks::TPortTask>(std::move(active_ids),tPort,
+                                                                   std::move(read_manager_),std::move(write_manager_));
 
         pool.AddTask(nPortTask);
         pool.AddTask(tPortTask);
 
-        while (pool.GetTaskStatus(nPortTask) != thread_pool::Task::Status::DONE ||
-               pool.GetTaskStatus(tPortTask) != thread_pool::Task::Status::DONE) {
+        while (pool.GetTaskStatus(nPortTask) != thread_pool::Task::Status::RUNNING ||
```

Comment:
Wrong logic here. This should be `==` now, not `!=`.

Reply:
Duh. Whoops.